### PR TITLE
Onboarding

### PR DESCRIPTION
**Pull request by emilyreynolds89**
# Description
Fixed style in on boarding and saving user selected organization and tags for their preferences. 

Fixes # (issue)
Now saving user's selects in onboarding

## Type of change
Styling and saving users account & preferences

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Tested on emulator. Tried on restarted emulator, with new accounts. Logged selected and unselected buttons to make sure preferences are accurately saved. 

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: Events/app/src/main/java/com/dti/cornell/events/OnboardingActivity.java [hunk 1]
```diff
@@ -165,10 +201,17 @@ public void onClick(View view)
...
 					break;
 				case R.id.getStartedButton:
+					((OnboardingActivity) getActivity()).flipPage();
+					break;
 				case R.id.nextButton:
+					saveSelectedOrganizations(orgAdapter.getSelectedOrganizations());
+					((OnboardingActivity) getActivity()).flipPage();
+					break;
 				case R.id.doneButton:
+					saveSelectedTags(tagAdapter.getSelectedTags());
```
Comment (davidchuyaya) [inline Events/app/src/main/java/com/dti/cornell/events/OnboardingActivity.java]: hm that's a lot of copy-paste

File: Events/app/src/main/java/com/dti/cornell/events/ProfileFragment.java [hunk 2]
```diff
@@ -194,15 +194,15 @@ public void onClick(View view) {
 
 	public void configure(){
 		followingRecycler.setAdapter(new OrganizationAdapter(followingRecycler.getContext(),
-				Data.organizations().stream().filter((val)->OrganizationUtil.userIsFollowing(val.id)).collect(Collectors.toList()), false));
-		tagRecycler.setAdapter(new TagAdapter(tagRecycler.getContext(), ImmutableList.copyOf(TagUtil.getMostPopularTags(100)), false));
+				Data.organizations().stream().filter((val)->OrganizationUtil.userIsFollowing(val.id)).collect(Collectors.toList()), false, false));
```
Comment (davidchuyaya) [inline Events/app/src/main/java/com/dti/cornell/events/ProfileFragment.java]: Not sure why we're using Java 8 streams here
Remember that they're only supported from API 24 ([see here](https://developer.android.com/studio/write/java8-support#supported_features))
(btw I see that that is the minimum version supported in `build.gradle` now, presumably to allow Java 8. I think that kind of decision is best made after looking at stats of Android users at Cornell. API 21 should be min in my opinion but I'm old and that might be outdated info.)

File: Events/app/src/main/java/com/dti/cornell/events/TagAdapter.java [hunk 3]
```diff
@@ -15,17 +17,24 @@
...
+	private final boolean onboardingTag;
 
-	public TagAdapter(Context context, ImmutableList<Integer> tags, boolean selectable)
+	public TagAdapter(Context context, ImmutableList<Integer> tags, boolean selectable, boolean onboardingTag)
 	{
 		this.tags = tags;
 		this.selectable = selectable;
 		selected = new HashSet<>(tags.size());
 		inflater = (LayoutInflater) context.getSystemService(Context.LAYOUT_INFLATER_SERVICE);
+		this.onboardingTag = onboardingTag;
```
Comment (davidchuyaya) [inline Events/app/src/main/java/com/dti/cornell/events/TagAdapter.java]: probably can move this up 2 lines to keep things neat

File: Events/app/src/main/java/com/dti/cornell/events/TagAdapter.java [hunk 4]
```diff
@@ -57,6 +70,10 @@ public int getItemCount()
 		return tags.size();
 	}
 
+	public Set<Integer> getSelectedTags() {
```
Comment (davidchuyaya) [inline Events/app/src/main/java/com/dti/cornell/events/TagAdapter.java:73]: Not sure why it's private if you're just making a getter for it (which allows anyone to mutate the set anyway)

File: Events/app/src/main/java/com/dti/cornell/events/TagViewHolder.java [hunk 5]
```diff
@@ -15,15 +18,21 @@
 {
 	private final Context context;
 	private final Drawable background;
+	private final Integer textColor;
```
Comment (davidchuyaya) [inline Events/app/src/main/java/com/dti/cornell/events/TagViewHolder.java]: why not `int`?

File: Events/app/src/main/java/com/dti/cornell/events/TagViewHolder.java [hunk 6]
```diff
@@ -48,10 +63,13 @@ public void onClick(View v)
 		//TODO Check to see if the Activity name matches onboarding's activity name and don't open if so
 		//TODO perhaps use the fact that you know the activity is onboarding to save the clicked
 		//TODO tags in onboarding.
-		if(((Activity)v.getContext()).getLocalClassName().equalsIgnoreCase("Onboarding")){
+		Log.i("tag viewholder", ((Activity)v.getContext()).getLocalClassName());
+		EventBusUtils.SINGLETON.post(new EventBusUtils.TagSelected(id, position));
+		if(((Activity)v.getContext()).getLocalClassName().equalsIgnoreCase("OnboardingActivity")){
```
Comment (davidchuyaya) [inline Events/app/src/main/java/com/dti/cornell/events/TagViewHolder.java]: this is an atrocity

File: Events/app/src/main/res/layout/item_onboarding_organization.xml [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+    android:background="@drawable/bg_round_button_white"
+    android:layout_width="match_parent"
+    android:layout_height="wrap_content"
+    android:paddingTop="2dp"
+    android:paddingBottom="2dp">
+
+    <TextView
+        android:id="@+id/name"
+        style="@style/OrgName"
+        android:layout_width="300dp"
```
Comment (davidchuyaya) [inline Events/app/src/main/res/layout/item_onboarding_organization.xml]: bad idea

File: Events/app/src/main/res/layout/item_onboarding_organization.xml [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+        android:layout_height="wrap_content"
+        android:layout_marginStart="12dp"
+        android:layout_marginEnd="12sp"
```
Comment (davidchuyaya) [inline Events/app/src/main/res/layout/item_onboarding_organization.xml]: dp

File: Events/app/src/main/res/layout/item_onboarding_organization.xml [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+
+    <androidx.cardview.widget.CardView
+        android:id="@+id/toolbarImageContainer"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:layout_marginBottom="8dp"
+        android:layout_marginStart="12dp"
+        android:layout_marginTop="8dp"
+        android:background="@color/unloadedGray"
+        app:cardCornerRadius="25dp"
```
Comment (davidchuyaya) [inline Events/app/src/main/res/layout/item_onboarding_organization.xml:35]: seems arbitrary (keep it for now, but ask designers why)

File: Events/app/src/main/res/layout/item_tag_onboarding.xml [hunk 10]
```diff
@@ -0,0 +1,17 @@
+<?xml version="1.0" encoding="utf-8"?>
+<TextView xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:id="@+id/tag_name_onboarding"
+    style="@style/P1"
+    android:layout_width="wrap_content"
+    android:layout_height="36dp"
```
Comment (davidchuyaya) [inline Events/app/src/main/res/layout/item_tag_onboarding.xml]: bad idea

File: Events/app/src/main/res/layout/onboarding_1.xml [hunk 11]
```diff
@@ -42,10 +42,12 @@
 	<Button
 		android:id="@+id/getStartedButton"
 		style="@style/mainButton"
-		android:layout_width="wrap_content"
+		android:layout_width="160dp"
```
Comment (davidchuyaya) [inline Events/app/src/main/res/layout/onboarding_1.xml]: not the best idea

File: Events/app/src/main/res/layout/onboarding_4.xml [hunk 12]
```diff
@@ -44,25 +44,29 @@
 		android:id="@+id/tagRecycler"
 		android:layout_width="0dp"
 		android:layout_height="0dp"
-		android:layout_marginBottom="@dimen/spacing_3l"
+		android:layout_marginBottom="@dimen/spacing_xl"
 		android:layout_marginTop="@dimen/spacing_xl"
 		android:orientation="vertical"
+		android:paddingBottom="@dimen/spacing_5l"
+		android:clipToPadding="false"
+		android:paddingLeft="1dp"
```
Comment (davidchuyaya) [inline Events/app/src/main/res/layout/onboarding_4.xml:52]: ?

**Review by davidchuyaya — CHANGES_REQUESTED** [review]

File: Events/app/src/main/java/com/dti/cornell/events/TagViewHolder.java (hunk 6 above)
Comment (davidchuyaya) [reply Events/app/src/main/java/com/dti/cornell/events/TagViewHolder.java]: not your fault but maybe don't keep it

File: Events/app/src/main/res/layout/onboarding_4.xml (hunk 12 above)
Comment (emilyreynolds89) [reply Events/app/src/main/res/layout/onboarding_4.xml:52]: the leftmost items in the recycler view were slightly cut off on the left without the left padding

**Comment by emilyreynolds89** [comment]
Reviewed change requests & updated onboarding branch (accidentally pushed with same commit message as last, but code is updated from last commit)

File: Events/app/src/main/java/com/dti/cornell/events/MainActivity.java [hunk 13]
```diff
@@ -74,14 +75,14 @@ protected void onCreate(Bundle savedInstanceState)
 		super.onCreate(savedInstanceState);
 		setContentView(R.layout.activity_main);
 
-//		EventBusUtils.SINGLETON.register(this);
+		//EventBusUtils.SINGLETON.register(this);
```
Comment (SamChou19815) [inline Events/app/src/main/java/com/dti/cornell/events/MainActivity.java]: Don't leave commented code in the codebase

File: Events/app/src/main/java/com/dti/cornell/events/TagViewHolder.java (hunk 6 above)
Comment (davidchuyaya) [reply Events/app/src/main/java/com/dti/cornell/events/TagViewHolder.java]: ^ Let's get this fixed (just do a simple `instanceof`, which is janky enough)

**Review by davidchuyaya — CHANGES_REQUESTED** [review]

**Review by SamChou19815 — APPROVED** [review]

**Review by shichong-97 — APPROVED** [review]

**Comment by SamChou19815** [comment]
@shichong-97 @emilyreynolds89 Is this ready? Looks like it's impossible to dismiss @davidchuyaya 's review. When it's ready, I might need to temporary disable the branch protection to let this PR get through.

**Pull request closed without merging**